apps/freeablo/faworld/itembonus.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <cstdint>
+
+namespace DiabloExe
+{
+    class BaseItem;
+}
+
+namespace FAWorld
+{
+    /// ItemBonus is a class for storing addable properties of items which later applied to a player
+    /// It was decided to not use enum indexed vector for it because some of stats should be summed as a bitmask
+    /// In case of performance problem sum of ItemBonus'es can later be cached and updated only in case if equipment is changed
+    class ItemBonus
+    {
+        using selfType = ItemBonus;
```

Comment:
why not just write ItemBonus?

Reply:
Well it's matter of preference, I prefer to avoid repeating type's name too much, so it would be easier to move function to another class for example. I also used to name this typedef just `self` which makes it much shorter but I've seen some people complaining about it so I've tried adding `type` in this case.